### ISSUE TITLE
Crash on very large key expiration

### ISSUE THREAD

**Issue by fernandomacho** [dragonflydb/dragonfly#1362]
Yesterday I changed the expiration of about 200,000 keys, more or less they were all expiring at the same time. When it reached expiration time for the keys dragonfly crashed.

What you see in the syslog is:
Jun 5 23:45:34 ovh-innova dragonfly[3085274]: F20230605 23:45:34.081375 3085276 epoll_socket.cc:274] sock[51] Unexpected error 110/Connection timed out
Jun 5 23:45:34 ovh-innova dragonfly[3085274]: *** Check failure stack trace: ***
Jun 5 23:45:34 ovh-innova dragonfly[3085274]: @0x55ae649a6ce3 (unknown)
Jun 5 23:45:34 ovh-innova dragonfly[3085274]: @0x55ae6499f4a7 (unknown)
Jun 5 23:45:34 ovh-innova dragonfly[3085274]: @0x55ae649a0e2f (unknown)
Jun 5 23:45:34 ovh-innova dragonfly[3085274]: @ 0x55ae64518a6c (unknown)
Jun 5 23:45:34 ovh-innova dragonfly[3085274]: @ 0x55ae64511372 (unknown)
Jun 5 23:45:34 ovh-innova dragonfly[3085274]: @0x55ae64582c9b (unknown)
Jun 5 23:45:34 ovh-innova dragonfly[3085274]: @0x55ae6451fde2 (unknown)
Jun 5 23:45:34 ovh-innova dragonfly[3085274]: @ 0x55ae64520167 (unknown)
Jun 5 23:45:34 ovh-innova dragonfly[3085274]: @0x55ae647d2e3f (unknown)
Jun 5 23:45:34 ovh-innova dragonfly[3085274]: *** SIGABRT received at time=1686001534 on cpu 0 ***
Jun 5 23:45:34 ovh-innova dragonfly[3085274]: PC: @ 0x7fac6ac3200b (unknown) raise
Jun 5 23:45:34 ovh-innova systemd[1]: dragonfly.service: Main process exited, code=killed, status=6/ABRT
Jun 5 23:45:34 ovh-innova systemd[1]: dragonfly.service: Failed with result 'signal'.
Jun 5 23:45:34 ovh-innova systemd[1]: dragonfly.service: Scheduled restart job, restart counter is at 2.

 - OS: ubuntu 20.04.05
 - Kernel: 5.4.0-125-generic (this week end I try to update to 5.15 kernel)
 - Containerized?: Bare Metal
 - Dragonfly Version: 1.3.0

I have 3 dragonfly on my installation: 1 master and 2 replicas with Redis-sentinel 
The machines have 256Gb (master) and 192Gb (replicas) of ram

The startup command is (flag file content)

--pidfile=/var/run/dragonfly/dragonfly.pid
--log_dir=/var/log/dragonfly
--dir=/var/lib/dragonfly
--version_check=true
--port=6392
--dbnum=24
--hz=1
--proactor_threads=8
--save_schedule=*:00
--keys_output_limit=12288
--dbfilename=dump-{timestamp}

**Comment by romange** [comment]
@fernandomacho is it reproducible,  i.e. does it happen consistently? 

Did you succeed to failover to another replica after the crash?

**Comment by romange** [comment]
Also, which process has crashed? The replica or the master?

**Comment by fernandomacho** [comment]
After master crash, the master crashed remain as master ¿?

It has not happened again but a possible simulation could be to have a database with about 700,000 string-type keys, in my case I store jsons in those keys, of those 700,000 keys make about 10,000 start with "1:xxxxxx", for For example, another 10,000 that begin with "2:xxxxxx" and so on until the 200,000 keys to expire have been "located"
and run a command like:

```
for number in $(seq 1 20)
do
    for i in $(redis-cli -n 8 KEYS "$number:*")
    do 
       redis -cli -p 6392 -n 8 expire $i 120 
   done
done
```

Be careful what is in the config file so that it is able to return all the expected keys (--keys_output_limit=10000)

**Comment by romange** [comment]
I actually know what the problem is. Will issue the fix for the next version, thanks!

**Comment by romange** [comment]
what is your client stack  (language, client library) ?

**Comment by fernandomacho** [comment]
Php8.2 and php redis compile module

**Comment by romange** [comment]
Should be fixed in v1.4.0
